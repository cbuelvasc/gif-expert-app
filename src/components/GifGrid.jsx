import PropTypes from "prop-types";
import { GifItem, LoadingGifs } from "./";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h2 className="text-2xl font-bold tracking-tight text-gray-900 mt-6">
        {category}
      </h2>
      {isLoading && <LoadingGifs />}
      <div className="mt-6 grid grid-cols-1 gap-x-6 gap-y-10 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
