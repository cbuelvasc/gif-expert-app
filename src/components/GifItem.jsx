import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="group relative shadow-lg">
      <div className="aspect-h-1 aspect-w-1 w-full overflow-hidden rounded-md bg-gray-200 lg:aspect-none group-hover:opacity-75 lg:h-80 ">
        <img
          src={url}
          alt={title}
          className="h-full w-full object-cover object-center lg:h-full lg:w-full"
        />
      </div>
      <div className="mt-4 flex justify-between">
        <div>
          <h3 className="text-sm text-gray-700 pl-2 ">
            <a href="#">
              <span aria-hidden="true" className="absolute inset-0" />
              {title}
            </a>
          </h3>
        </div>
      </div>
      <div className="mt-4 flex justify-between"></div>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
