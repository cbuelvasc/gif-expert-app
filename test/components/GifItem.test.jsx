/* eslint-disable no-undef */

import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Testing for <GifItem />", () => {
  const title = "A title";
  const url = "https://localhost/image.jpg";

  test("should be matched with the snapshot", () => {
    const {container} = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should display the image with the URL and ALT indicate", () => {
    render(<GifItem title={title} url={url} />);

    const{src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should display the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    const h3 = screen.getByText(title);
    expect(h3).toBeTruthy();
  });
});
