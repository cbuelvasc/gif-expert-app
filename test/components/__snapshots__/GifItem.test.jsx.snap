// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for <GifItem /> should be matched with the snapshot 1`] = `
<div>
  <div
    class="group relative shadow-lg"
  >
    <div
      class="aspect-h-1 aspect-w-1 w-full overflow-hidden rounded-md bg-gray-200 lg:aspect-none group-hover:opacity-75 lg:h-80 "
    >
      <img
        alt="A title"
        class="h-full w-full object-cover object-center lg:h-full lg:w-full"
        src="https://localhost/image.jpg"
      />
    </div>
    <div
      class="mt-4 flex justify-between"
    >
      <div>
        <h3
          class="text-sm text-gray-700 pl-2 "
        >
          <a
            href="#"
          >
            <span
              aria-hidden="true"
              class="absolute inset-0"
            />
            A title
          </a>
        </h3>
      </div>
    </div>
    <div
      class="mt-4 flex justify-between"
    />
  </div>
</div>
`;
